assets/js/modules/analytics-4/datastore/accounts.js

```diff
@@ -41,6 +43,35 @@ const baseInitialState = {
 };
 
 const baseActions = {
+	/**
+	 * Gets an ID of an account that has a matching property.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} The account ID on success, otherwise NULL.
+	 */
+	*matchAccountID() {
+		const registry = yield commonActions.getRegistry();
+		const accounts = yield Data.commonActions.await(
+			registry.__experimentalResolveSelect( STORE_NAME ).getAccountSummaries()
+		);
+
+		if ( ! Array.isArray( accounts ) || accounts.length === 0 ) {
+			return null;
+		}
+
+		const url = registry.select( CORE_SITE ).getReferenceSiteURL();
+		const ga4PropertyIDs = ( Array.isArray( accounts ) ? accounts : [] ).reduce(
+			( acc, { propertySummaries } ) => [ ...acc, ...propertySummaries.map( ( { _id } ) => _id ) ],
+			[],
+		);
+
+		const property = yield Data.commonActions.await(
+			registry.dispatch( STORE_NAME ).matchPropertyByURL( ga4PropertyIDs, url )
+		);
+
+		return property?._accountID;
```

Comment:
How about we return the whole property here, that way we can select it directly rather than matching it again with the `accountID`? Then this could be renamed (and probably moved) to `matchProperty` or maybe `findMatchedProperty`.

Reply:
Yep, makes sense. Updated.